src/test/skript/tests/syntaxes/expressions/ExprBeehiveFlower.sk

```diff
@@ -0,0 +1,14 @@
+test "beehive flower":
+	set {_old} to block at test-block
+	set {_old2} to block at {_old} ~ vector(1, 0, 0)
+	set block at test-block to beehive
+	set {_hive} to block at test-block
+	set block at {_old2} to flower
+	set {_flower} to block at {_old2}
+	set the target flower of {_hive} to {_flower}
+	assert the block at (target flower of {_hive}) is {_flower} with "Beehive flower target did not change"
+	clear the target flower of {_hive}
+	assert the target flower of {_hive} is not set with "Beehive flower target was not cleared"
+
+	set block at test-block to {_old}
+	set block at {_old2} to {_old2}
```

Comment:
old and old2 are block references, so this literally just sets them to themselves, it doesn't restore previous state.